src/wp-includes/rest-api.php

```diff
@@ -1575,6 +1575,208 @@ function rest_find_matching_pattern_property_schema( $property, $args ) {
 	return null;
 }
 
+/**
+ * Gets the error of combining operation.
+ *
+ * @since 5.6.0
+ *
+ * @param array  $value  The value to validate.
+ * @param string $param  The parameter name, used in error messages.
+ * @param array  $errors The errors array, to search for possible error.
+ * @return WP_Error      The combining operation error.
+ */
+function rest_get_combining_operation_error( $value, $param, $errors ) {
+	// If there is only one error, simply return it.
+	if ( 1 === count( $errors ) ) {
+		$position = $errors[0]['index'];
+		$reason   = $errors[0]['error_object']->get_error_message();
+
+		/* translators: 1: Parameter, 2: Reason. */
+		return new WP_Error(
+			'rest_invalid_param',
+			sprintf( __( '%1$s does not match any of the expected formats. Reason: %2$s' ), $param, $reason ),
+			array( 'position' => $position )
+		);
+	}
+
+	// Filter out all errors related to type validation.
+	$filtered_errors = array();
+	foreach ( $errors as $error ) {
+		if (
+			'rest_invalid_type' !== $error['error_object']->get_error_code() ||
+			$param !== $error['error_object']->get_error_data()['param']
+		) {
+			$filtered_errors[] = $error;
+		}
+	}
+
+	// If there is only one error left, simply return it.
+	if ( 1 === count( $filtered_errors ) ) {
+		$position = $filtered_errors[0]['index'];
+		$reason   = $filtered_errors[0]['error_object']->get_error_message();
+
+		/* translators: 1: Parameter, 2: Reason. */
+		return new WP_Error(
+			'rest_invalid_param',
+			sprintf( __( '%1$s does not match any of the expected formats. Reason: %2$s' ), $param, $reason ),
+			array( 'position' => $position )
+		);
+	}
+
+	// If there are only errors related to object validation, try choosing the most appropriate one.
+	if ( count( $filtered_errors ) > 1 && 'object' === $filtered_errors[0]['schema']['type'] ) {
```

Comment:
```suggestion
	if ( 'object' === $filtered_errors[0]['schema']['type'] ) {
```

I think this conditional is no longer needed because it is covered in the previous if statement?

Reply:
There may be zero errors after filtering, so we need to do this check.

Reply:
Ah, thanks for clarifying. Sorry for missing that. What error message do we end up returning in that case, do we have a test case?

Reply:
If there are no errors, we end up with the most generic message (see lines 1687-1701).
`test_combining_operation_error_message` checks all variants of messages that this function generates.